web-api/src/embedding.rs

```diff
@@ -17,14 +17,26 @@ use xayn_ai_bert::{AvgEmbedder, Config as EmbedderConfig, NormalizedEmbedding};
 
 use crate::{app::SetupError, error::common::InternalError, utils::RelativePathBuf};
 
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum Config {
+    Pipeline(Pipeline),
```

Comment:
:art: we could also make this a structured variant like
```suggestion
    Pipeline {
        directory: RelativePathBuf,
        token_size: usize,
    },
```

Reply:
I just tried to use a structured variant and i ran into the problem that cannot use #[serde(default)] on the enum. this becomes an issue when running the integration tests 
```
Error: invalid type: found string "/Users/debug/projects/xayn_discovery_engine/assets/xaynia_v0002", expected struct RelativePathBuf for key "default.embedding" in TOML source string
```
`cargo test -p xayn-web-api@0.2.0 --test document_candidates -- --nocapture`

Reply:
I also tried to implement the alternative solution described [here](https://github.com/serde-rs/serde/issues/1799#issuecomment-624978919) to define a default but it don't not work in combination with having a default on the config enum itself.

Reply:
yeah, unfortunately `#[serde(default)]` doesn't extend to structured enum variants. we could work around that in some way, but the current solution should be ok.